### PR TITLE
HSEARCH-2744 Improve the Simple Query String documentation

### PR DESCRIPTION
**Pull request by gsmet**
* https://hibernate.atlassian.net/browse/HSEARCH-2744

### CODE REVIEW

File: documentation/src/main/asciidoc/query.asciidoc [hunk 1]
```diff
@@ -289,25 +289,54 @@ Query luceneQuery = mythQB.keyword()
...
     .simpleQueryString()
     .onField("history")
-    .matching("(storm OR hurricane) AND \"falling branches\"")
+    .matching("storm")
     .createQuery();
 ----
 ====
 
-By default, a simple query string query uses the OR operator if the operator is not explicitly defined.
+By default, a Simple Query String query uses the OR operator if the operator is not explicitly defined.
```
Comment (Sanne) [inline documentation/src/main/asciidoc/query.asciidoc]: +1 let's keep it consistently capitalized to clarify that it's the proper name of the feature.

File: documentation/src/main/asciidoc/query.asciidoc [hunk 2]
```diff
@@ -339,12 +368,12 @@ Query luceneQuery = mythQB
 ----
 ====
 
-The above query returns results containing "storm" and "tree" either in "name", "history" or "description".
+The above query returns results containing `storm` and `tree` either in `name`, `history` or `description`.
 
-The syntax supported by the SimpleQueryParser includes the following operators: boolean (AND, OR, NOT),
+The syntax supported by the `SimpleQueryParser` includes the following operators: boolean (AND `+`, OR `|`, NOT `-`),
 prefix, phrase, precedence, fuzzy and near operator for phrase queries.
```
Comment (Sanne) [inline documentation/src/main/asciidoc/query.asciidoc]: could you list also how to do "prefix, phrase, precedence, fuzzy " ?

File: documentation/src/main/asciidoc/query.asciidoc [hunk 3]
```diff
@@ -339,12 +368,12 @@ Query luceneQuery = mythQB
 ----
 ====
 
-The above query returns results containing "storm" and "tree" either in "name", "history" or "description".
+The above query returns results containing `storm` and `tree` either in `name`, `history` or `description`.
 
-The syntax supported by the SimpleQueryParser includes the following operators: boolean (AND, OR, NOT),
+The syntax supported by the `SimpleQueryParser` includes the following operators: boolean (AND `+`, OR `|`, NOT `-`),
```
Comment (yrodiere) [inline documentation/src/main/asciidoc/query.asciidoc]: It's definitely better, I'm a bit worried that users will understand that `AND`, `OR`, etc. can be used as operators in queries, while in fact they cannot.
Maybe something like "(AND using `+`, OR using `|`, NOT using `-`)" would avoid any risk of confusion?

Comment (gsmet) [inline documentation/src/main/asciidoc/query.asciidoc]: I was just changing it following the above remark of Sanne (and the use of a list to present each option). :)

**Comment by gsmet** [comment]
@Sanne @yrodiere force pushed an update addressing your comments on the operator list. It should be more clear now.

**Comment by yrodiere** [comment]
Nice! Since @Sanne's remarks have been addressed too, I merged it.

**Pull request merged**